flink-core/src/main/java/org/apache/flink/configuration/PipelineOptions.java

```diff
@@ -165,6 +166,14 @@ public class PipelineOptions {
                             "Register a custom, serializable user configuration object. The configuration can be "
                                     + " accessed in operators");
 
+    public static final ConfigOption<Map<String, Integer>> PARALLELISM_OVERRIDES =
+            key("pipeline.jobvertex-parallelism-overrides")
+                    .mapType(Integer.class)
+                    .defaultValue(Collections.emptyMap())
+                    .withDescription(
+                            "A parallelism override map (jobVertexId -> parallelism) which will be used to update"
+                                    + " the parallelism of the corresponding job vertices of submitted JobGraphs.");
```

Comment:
can we have a description that exposes less internals?

job graph -> application
job vertices -> operator

Reply:
We could but since the overrides are derived from the JobGraph and that's an implementation detail per-se, I don't think we can disguise that if we add this option.

A job vertex hosts one or more operators, so calling it operator would be incorrect.